contrib/document_constants.py

```diff
@@ -0,0 +1,280 @@
+        measure,
+    )
+    if not omit_empty_columns:
+        def filter_visible_columns(seq):
+            return tuple(seq)
+    else:
+        visible_columns = [bool(ml) for ml in max_lengths]
+        def filter_visible_columns(seq):
+            return tuple(x for i, x in enumerate(seq) if visible_columns[i])
+
+    max_lengths = get_max_values(
+        max_lengths,
+        measure_items(header, measure),
+    )
+
+    visible_columns_lengths = filter_visible_columns(max_lengths)
+    fmt = create_table_format(visible_columns_lengths)
+    ret = [
+        fmt % filter_visible_columns(header),
+        create_table_header_separator(visible_columns_lengths),
+    ]
+    for r in rows:
+        ret.append(fmt % filter_visible_columns(r))
+    return "\n".join(ret)
+
+
+class DocumentEnum:
+    cls: Enum
+    table_header = ("Symbol", "Value", "Domain", "Description")
+
+    def __init__(self, cls: Enum) -> None:
+        self.cls = cls
+
+    @property
+    def name(self):
+        return self.cls.__name__
+
+    def __str__(self) -> str:
+        lines = []
+        if self.cls.__doc__ and self.cls.__doc__ != GenericEnum.__doc__:
+            doc = self.cls.__doc__.splitlines()
+            if len(doc) > 2 and doc[0] and not doc[1]:
+                lines.append(f"## {self.name} ({doc[0].strip()})")
+                doc.pop(0)
+                doc.pop(0)
+            else:
+                lines.append(f"## {self.name}")
+
+            lines.append(dedent("\n".join(doc)))
+            lines.append("")
+        else:
+            lines.append(f"## {self.name}")
+
+        rows = []
+        for m in self.cls.__members__.values():
+            v = m.value
+            if isinstance(v, constants.VSSConstant):
+                rows.append((f"`{m.name}`", v, v.domain, v.description))
+            else:
+                rows.append((f"`{m.name}`", v, '', ''))
+
+        lines.append(create_table(self.table_header, rows))
+        lines.append("")
+        return "\n".join(lines)
+
+
+
+def collect() -> Sequence[DocumentEnum]:
+    collected: List[DocumentEnum] = []
+    for name in constants.__all__:
+        element = getattr(constants, name)
+        if issubclass(element, Enum):
+            collected.append(DocumentEnum(element))
+    return collected
+
+
+def get_args() -> argparse.ArgumentParser:
+    ap = argparse.ArgumentParser(
+        description="Generate MarkDown documentation based on code constants"
+    )
+    ap.add_argument(
+        "--name", "-n",
+        help="If provided, restrict the name of constant to document.",
+    )
+    ap.add_argument(
+        "--output-file", "-o",
+        help="File to write (defaults to stdout)",
+        type=argparse.FileType("w"),
+        default=sys.stdout,
+    )
+    return ap
+
+
+def main():
+    ap = get_args()
+    args = ap.parse_args()
+
+    out = args.output_file
+    collected = collect()
+    for doc in collected:
+        if args.name and args.name != doc.name:
```

Comment:
@gunnarx I still keep this `--name` and check so we can use the same script to generate different markup files for a single type, like we have:
- https://github.com/GENIVI/vehicle_signal_specification/blob/master/docs-gen/content/rule_set/data_entry/data_types.md
- https://github.com/GENIVI/vehicle_signal_specification/blob/master/docs-gen/content/rule_set/data_entry/data_unit_types.md

(Note: I just realized/noticed that the `data_types.md` contains some different columns, so shall we keep this tool I'll update the `VSSConstant` and this tool to handle different columns, likely `VSSUnitConstant` and `VSSDataTypeConstant`)